docs/tutorials/in-the-cloud.qmd

```diff
@@ -0,0 +1,302 @@
+upload that to GCP for processing. Once the workload operator is finished, the
+parties are able to retrieve their linkage results.
+
+### Workload author
+
+The workload **author** is responsible for building a Docker image containing
+the cloud-based linkage code and uploading it to a GCP Artifact Registry. This
+image is the workload to be run by the operator.
+
+### Workload operator
+
+The workload **operator** runs the linkage itself using some embedded data from
+the parties and an image from the author. They are responsible for setting up
+and running a
+[Confidential Space](https://cloud.google.com/docs/security/confidential-space)
+in which to perform the linkage. This setting ensures that nobody ever has
+access to all the data at once, and that the data can only be accessed via the
+linkage code itself.
+
+
+## Creating your GCP projects
+
+Once you have decided who will be playing which role(s), you need to decide on
+a naming structure and make some GCP projects. You will need a project for each
+member of the linkage project - not one for each role. The names of these
+projects will be used throughout the cloud implementation, from configuration
+files to buckets. As such, they need to be descriptive and unique.
+
+::: {.callout-warning}
+Since Google Cloud bucket names must be
+[globally unique](https://cloud.google.com/storage/docs/buckets#naming), we
+highly recommend using a hash in your project names to ensure that they are
+unique.
+
+Our aim is to create a globally unique name (and thus ID) for each project.
+:::
+
+For example, say the US Census Bureau and UK Office for National Statistics
+(ONS) are looking to link some data on ex-patriated residents with PPRL. Then
+they might use `us-cb` and `uk-ons` as their party names, which are succinct
+and descriptive. However, they are generic and rule out future PPRL projects
+with the same names.
+
+As a remedy, they could make a hash of their project description to create an
+identifier:
+
+```bash
+$ echo -n "pprl us-cb uk-ons ex-pats-analysis" | sha256sum
+d59a50241dc78c3f926b565937b99614b7bb7c84e44fb780440718cb2b0ddc1b  -
+```
+
+This is very long. You might only want to use the first few characters of this
+hash. Note that Google Cloud bucket names also can't be more than 63 characters
+long without dots.
+
+You can trim it down like so:
+
+```bash
+$ echo -n "pprl us-cb uk-ons ex-pats-analysis" | sha256sum | cut -c 1-7
+d59a502
+```
+
+So, our names would be: `uk-ons-d59a502`, `us-cb-d59a502`. If they had a
+third-party linkage administrator (authoring and operating the workload), they
+would have a project called something like `admin-d59a502`.
+
+
+## Setting up your projects
+
+Once you have decided on a naming structure, it is time to create the GCP
+projects. Each project will need specific Identity and Access Management (IAM)
+roles granted to them by the project owner's GCP Administrator. Which IAM roles
```

Comment:
Could we include a `gcloud` snippet here to show them how to assign roles?

Reply:
I am not sure what the `gcloud` commands would be here, but I can try and dig something out

Reply:
Fixed by 101fbf2